### ISSUE TITLE
HF solution depends on available memory for 2D, GDF/MDF systems

### ISSUE THREAD

**Issue by Majokux** [pyscf/pyscf#594]
Using this minimal example, mostly taken from `example/pbc/32`:

```from pyscf.pbc import df as pdf
from pyscf.pbc import gto as pbcgto
from pyscf.pbc import scf as pbchf

Lz = 25 # Smallest Lz value for ~1e-6 convergence in absolute energy
a = 1.42 # bond length in graphene

cell = pbcgto.Cell()
cell.build(unit = 'B',
           a = [[4.6298286730500005, 0.0, 0.0], [-2.3149143365249993, 4.009549246030899, 0.0], [0.0, 0.0, Lz]],
           atom = 'C 0 0 0; C 0 2.67303283 0',
           dimension=2,
           pseudo = "gth-pade",
           verbose = 4,
           precision = 1e-6,
           basis='gth-szv',
           #max_memory=1,
           )

mf = pbchf.RHF(cell)
mf = mf.density_fit()
mf.conv_tol = 1e-6
e_mf = mf.kernel()
```

I converge to `converged SCF energy = -10.7012224101449`
On the other hand, if I uncomment the `max_memory = 1,` line, I converge to
`converged SCF energy = -14.6469819237595` (This is just a minimal example - the way I found this behaviour was that the mean-field solution was different depending on whether an unrelated CCSD calculation was performed or not).

This seem to only happen for 2D systems with GDF and MDF. Taking a wild guess, is the negative mode in the three center integrals (for 2D systems) not correctly stored/loaded if not enough memory is available to store them incore?

More info:
```System: uname_result(system='Linux', node='zombie', release='4.4.0-174-generic', version='#204-Ubuntu SMP Wed Jan 29 06:41:01 UTC 2020', machine='x86_64', processor='x86_64')  Threads 8
Python 3.5.2 (default, Apr 16 2020, 17:47:17)
[GCC 5.4.0 20160609]
numpy 1.15.4  scipy 1.1.0
Date: Tue Apr 28 16:16:01 2020
PySCF version 1.7.1
PySCF path  /apps/pyscf/1.7.1/pyscf
GIT HEAD      d11041db344dbf798445e574a6cfe72e16f3549d```

**Comment by fishjojo** [comment]
I couldn't reproduce the error with the latest release.
[594_1.txt](https://github.com/pyscf/pyscf/files/4548781/594_1.txt)
[594.txt](https://github.com/pyscf/pyscf/files/4548782/594.txt)

**Comment by Majokux** [comment]
Your output seems to indicate that you set `cell.dimension` to 3?

**Comment by fishjojo** [comment]
ok, I can reproduce the error now. I'll take a look.

**Comment by fishjojo** [comment]
fixed in 36731ed
